src/plugins/intel_gpu/src/graph/col_to_im.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "col_to_im_inst.h"
+#include "col2im_shape_inference.hpp"
+
+#include "primitive_type_base.h"
+#include "intel_gpu/runtime/error_handler.hpp"
+#include "json_object.h"
+#include <string>
+
+namespace cldnn {
+GPU_DEFINE_PRIMITIVE_TYPE_ID(col_to_im)
+
+layout col_to_im_inst::calc_output_layout(col_to_im_node const& node, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<col_to_im>();
+
+    auto input_layout = impl_param.get_input_layout();
+    auto input_format = input_layout.format;
+
+    // TODO : do sth here for col2im.(Copied dummy from depth_to_space)
+    auto out_size = input_layout.get_tensor();
+    if (format::spatial_num(input_layout.format) == 3) {
+        const size_t feature = input_layout.feature() / (desc->kernel_shape[0] * desc->kernel_shape[1]);
+        const size_t z = 1;
+        const size_t y = desc->output_shape[1];
+        const size_t x = desc->output_shape[0];
+        out_size = tensor(TensorValue(input_layout.batch()), TensorValue(feature), TensorValue(x), TensorValue(y), TensorValue(z));
+    } else {
+        const size_t feature = input_layout.feature() / (desc->kernel_shape[0] * desc->kernel_shape[1]);
+        const size_t y = desc->output_shape[1];
+        const size_t x = desc->output_shape[0];
+        out_size = tensor(TensorValue(input_layout.batch()), TensorValue(feature), TensorValue(x), TensorValue(y));
+    }
+
+    if (impl_param.has_fused_primitives()) {
+        input_layout.data_type = impl_param.get_output_element_type();
+    }
+
+    return layout{input_layout.data_type, input_format, out_size};
+}
+
+template<typename ShapeType>
+std::vector<layout> col_to_im_inst::calc_output_layouts(col_to_im_node const& node, kernel_impl_params const& impl_param) {
```

Comment:
Please check this function implementation. As stated below, it is a quick and dirty implementation. Proper implementation should use `shape_infer` function to calculate output_shapes.

Reply:
Fixed.